src/crontab.c

```diff
@@ -500,8 +500,8 @@ static int backup_crontab(const char *crontab_path) {
 	const char *env_value;
 	char backup_dir[MAX_FNAME], backup_path[MAX_FNAME];
 	int ch = '\0';
-	FILE *crontab_file;
-	FILE *backup_file;
+	FILE *crontab_file = NULL;
+	FILE *backup_file = NULL;
```

Comment:
The initialization should not be necessary as we never call fclose with these `FILE *` variables uninitialized. Also calling fclose on NULL is afaik not allowed so please check whether this isn't the case. 

There are missing fclose calls on lines 566 and 593 though, could you please add them?

Reply:
> Also calling fclose on NULL is afaik not allowed so please check whether this isn't the case.

TIL; I assumed that passing `NULL` resumed in no-op, but this is apparently not the case according to standard. Which also means that the cleanup section is wrong – if opening of the backup file fails, `fclose(backup_file)` is called with `backup_file == NULL`. I'll add guard checks for that.

> The initialization should not be necessary as we never call fclose with these FILE * variables uninitialized.

True; I'll drop them then.

> There are missing fclose calls on lines 566 and 593 though, could you please add them?

I'll take a look.

Reply:
> There are missing fclose calls on lines 566 and 593 though, could you please add them?

Well, the `exit()` calls there kill the process at that point; *any* open files will be closed by the OS. Are the calls really necessary?